stake-pool/program/tests/vsa_remove.rs

```diff
@@ -104,3 +148,449 @@ async fn test_remove_validator_stake_account() {
+        user_stake,
+        user_pool_account,
+        _,
+    ) = setup().await;
+
+    let wrong_stake_program = Keypair::new();
+
+    let new_authority = Pubkey::new_unique();
+    let mut transaction = Transaction::new_with_payer(
+        &[instruction::remove_validator_stake_account(
+            &id(),
+            &stake_pool_accounts.stake_pool.pubkey(),
+            &stake_pool_accounts.owner.pubkey(),
+            &stake_pool_accounts.withdraw_authority,
+            &new_authority,
+            &stake_pool_accounts.validator_stake_list.pubkey(),
+            &user_stake.stake_account,
+            &user_pool_account.pubkey(),
+            &stake_pool_accounts.pool_mint.pubkey(),
+            &spl_token::id(),
+            &wrong_stake_program.pubkey(),
+        )
+        .unwrap()],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[&payer, &stake_pool_accounts.owner], recent_blockhash);
+    let transaction_error = banks_client
+        .process_transaction(transaction)
+        .await
+        .err()
+        .unwrap();
+
+    match transaction_error {
+        TransportError::TransactionError(TransactionError::InstructionError(
+            _,
+            error,
+        )) => {
+            assert_eq!(error, InstructionError::IncorrectProgramId);
+        }
+        _ => panic!("Wrong error occurs while try to remove validator stake address with wrong stake program ID"),
+    }
+}
+
+#[tokio::test]
+async fn test_remove_validator_stake_account_with_wrong_token_program_id() {
+    let (
+        mut banks_client,
+        payer,
+        recent_blockhash,
+        stake_pool_accounts,
+        user_stake,
+        user_pool_account,
+        _,
+    ) = setup().await;
+
+    let wrong_token_program = Keypair::new();
+
+    let new_authority = Pubkey::new_unique();
+    let mut transaction = Transaction::new_with_payer(
+        &[instruction::remove_validator_stake_account(
+            &id(),
+            &stake_pool_accounts.stake_pool.pubkey(),
+            &stake_pool_accounts.owner.pubkey(),
+            &stake_pool_accounts.withdraw_authority,
+            &new_authority,
+            &stake_pool_accounts.validator_stake_list.pubkey(),
+            &user_stake.stake_account,
+            &user_pool_account.pubkey(),
+            &stake_pool_accounts.pool_mint.pubkey(),
+            &wrong_token_program.pubkey(),
+            &stake::id(),
+        )
+        .unwrap()],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[&payer, &stake_pool_accounts.owner], recent_blockhash);
+    let transaction_error = banks_client
+        .process_transaction(transaction)
+        .await
+        .err()
+        .unwrap();
+
+    match transaction_error {
+        TransportError::TransactionError(TransactionError::InstructionError(
+            _,
+            InstructionError::Custom(error_index),
+        )) => {
+            let program_error = error::StakePoolError::InvalidProgramAddress as u32;
```

Comment:
Shouldn't this be `IncorrectProgramId`? What is `new_authority` for?

Reply:
Same issue with the wrong error from the contract. And `new_authority` is a required param in removing stake account from the pool (it is the new pubkey to be set as Staker and Withdrawer).

Reply:
Oh right, I see, thanks for the explanation